aspnet/fundamentals/middleware.rst

```diff
@@ -144,20 +165,20 @@ ASP.NET ships with the following middleware components:
 Writing middleware
 ------------------
 
+The `CodeLabs middleware tutorial <https://github.com/Microsoft-Build-2016/CodeLabs-WebDev/tree/master/Module2-AspNetCore>`__ provides a good introduction to writing middleware.
```

Comment:
Let's put this in the related content section. These docs should be as stand-alone as possible.